engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/MockConfluentSchemaRegistryClientFactory.scala

```diff
@@ -8,33 +8,46 @@ import pl.touk.nussknacker.engine.kafka.KafkaConfig
 
 import scala.collection.mutable.ListBuffer
 
-class MockConfluentSchemaRegistryClientFactory(data: List[(String, Schema, Boolean)]) extends ConfluentSchemaRegistryClientFactory {
+class MockConfluentSchemaRegistryClientFactory(data: List[RegistryItem]) extends ConfluentSchemaRegistryClientFactory {
   def createSchemaRegistryClient(kafkaConfig: KafkaConfig): TypedConfluentSchemaRegistryClient = {
     val mockSchemaRegistryClient = new MockSchemaRegistryClient with ConfluentSchemaRegistryClient
     data.foreach(item => register(mockSchemaRegistryClient, item))
     mockSchemaRegistryClient
   }
 
-  private def register(mockSchemaRegistry: MockSchemaRegistryClient, item: (String, Schema, Boolean)): MockSchemaRegistryClient = {
-    val subject = item._1 + "-" + (if (item._3) "key" else "value")
-    mockSchemaRegistry.register(subject, item._2)
+  private def register(mockSchemaRegistry: MockSchemaRegistryClient, item: RegistryItem): MockSchemaRegistryClient = {
+    val subject = item.topic + "-" + (if (item.isKey) "key" else "value")
+    mockSchemaRegistry.register(subject, item.schema, item.version, -1)
     mockSchemaRegistry
   }
 }
 
 class MockConfluentSchemaRegistryClientFactoryBuilder {
-  private val registry: ListBuffer[(String, Schema, Boolean)] = ListBuffer()
+  private val registry: ListBuffer[RegistryItem] = ListBuffer()
 
-  def register(topic: String, schema: Schema, isKey: Boolean = false): MockConfluentSchemaRegistryClientFactoryBuilder = {
-    registry.append((topic, schema, isKey))
+  def register(topic: String, schema: Schema, version: Int, isKey: Boolean): MockConfluentSchemaRegistryClientFactoryBuilder = {
+    registry.append(RegistryItem(topic, schema, version, isKey))
     this
   }
 
-  def registerStringSchema(topic: String, schema: String, isKey: Boolean = false): MockConfluentSchemaRegistryClientFactoryBuilder = {
-    registry.append((topic, AvroUtils.createSchema(schema), isKey))
+  def register(topic: String, schema: String, version: Int, isKey: Boolean): MockConfluentSchemaRegistryClientFactoryBuilder = {
+    registry.append(RegistryItem(topic,schema, version, isKey))
     this
   }
 
   def build: MockConfluentSchemaRegistryClientFactory =
     new MockConfluentSchemaRegistryClientFactory(registry.toList)
 }
+
+case class RegistryItem(topic: String, schema: Schema, version: Int, isKey: Boolean, id: Int)
+
+object RegistryItem {
+  def apply(topic: String, schema: String, version: Int, isKey: Boolean): RegistryItem =
+    new RegistryItem(topic, AvroUtils.createSchema(schema), version, isKey, -1)
+
+  def apply(topic: String, schema: Schema, version: Int, isKey: Boolean): RegistryItem =
+    new RegistryItem(topic, schema, version, isKey, -1)
+
+  def apply(topic: String, schema: Schema, version: Int): RegistryItem =
```

Comment:
I don't like such many versions of `apply`. It i hard to pick the correct one. I think that when we cleanup `id` problem, then it will be much easier. Maybe will be enough to have only two: `key` and `value` factory methods?

Reply:
`id` isn't a problem.. It's base prop for mock register method.  I removed last one - we don't use it. Now we have only 2 methods.